### PR TITLE
CreateWorkoutDao attached

### PR DESCRIPTION
**Pull request by DmitrySergienko**

### CODE REVIEW

**Comment by DmitrySergienko** [comment]
Dear Colleagues please review before merge! I'm not confident with it.

File: app/src/main/java/com/example/fitnessconstructor/database/CreateWorkoutDao.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import androidx.room.*
+import com.example.fitnessconstructor.domain.entities.Exercise
+import kotlinx.coroutines.flow.Flow
+
+@Dao
+interface CreateWorkoutDao {
+
+    //create workout
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun createWorkout(workout_id: Int?)
```
Comment (MarDSoul) [inline app/src/main/java/com/example/fitnessconstructor/database/CreateWorkoutDao.kt:12]:
```
suspend fun createWorkout(workout: WorkoutEntity)
```

File: app/src/main/java/com/example/fitnessconstructor/database/CreateWorkoutDao.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    //editWorkout
+    @Update
+    suspend fun editWorkout(workout_id: Int?)
```
Comment (MarDSoul) [inline app/src/main/java/com/example/fitnessconstructor/database/CreateWorkoutDao.kt:16]:
```
suspend fun editWorkout(workout: WorkoutEntity)
```

File: app/src/main/java/com/example/fitnessconstructor/database/CreateWorkoutDao.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    //getAllExercises
+    @Query("SELECT * FROM all_exercises")
+    fun getExercise(): Flow<List<Exercise>>
+
+    //addExerciseToWorkout
+    @Insert(onConflict = OnConflictStrategy.REPLACE )
+    fun addExercise(exercise_id: Int?)
```
Comment (MarDSoul) [inline app/src/main/java/com/example/fitnessconstructor/database/CreateWorkoutDao.kt:24]:
```
suspend fun addExercise(exercise: WorkoutExercises)
```

File: app/src/main/java/com/example/fitnessconstructor/database/CreateWorkoutDao.kt [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun createWorkout(workout_id: Int?)
+
+    //editWorkout
+    @Update
+    suspend fun editWorkout(workout_id: Int?)
+
+    //getAllExercises
+    @Query("SELECT * FROM all_exercises")
+    fun getExercise(): Flow<List<Exercise>>
```
Comment (MarDSoul) [inline app/src/main/java/com/example/fitnessconstructor/database/CreateWorkoutDao.kt:20]: Can't understand what do you want

File: app/src/main/java/com/example/fitnessconstructor/database/CreateWorkoutDao.kt [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    //addExerciseToWorkout
+    @Insert(onConflict = OnConflictStrategy.REPLACE )
+    fun addExercise(exercise_id: Int?)
+
+    //removeExerciseFromWorkout
+    @Delete
+    fun deleteExerciseFromWorkout(exercise_id: Int?)
```
Comment (MarDSoul) [inline app/src/main/java/com/example/fitnessconstructor/database/CreateWorkoutDao.kt:28]:
```
fun deleteExerciseFromWorkout(exercise: WorkoutExercises)
```

File: app/src/main/java/com/example/fitnessconstructor/database/CreateWorkoutDao.kt [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+    //deleteWorkout
+    @Query("DELETE FROM workout")
+    fun deleteWorkout()
```
Comment (MarDSoul) [inline app/src/main/java/com/example/fitnessconstructor/database/CreateWorkoutDao.kt:32]: This fun clear all table

**Review by MarDSoul — CHANGES_REQUESTED** [review]

**Review by MarDSoul — APPROVED** [review]

**Review by Pushin1007 — APPROVED** [review]

**Pull request merged**